sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/CombineFns.java

```diff
@@ -674,6 +674,26 @@ public void encode(Object[] value, OutputStream outStream, Coder.Context context
       coders.get(lastIndex).encode(value[lastIndex], outStream, context);
     }
 
+    @Override
+    public long getEncodedElementByteSize(Object[] value) throws Exception {
+      if (value.length == 0) {
+        return 0;
+      }
+      long size = 0;
+      for (int i = 0; i < value.length; ++i) {
+        Coder<Object> objectCoder = coders.get(i);
+        if (objectCoder instanceof StructuredCoder) {
```

Comment:
I've seen this type of condition[here](https://github.com/apache/beam/blob/master/sdks/java/core/src/main/java/org/apache/beam/sdk/coders/LengthPrefixCoder.java#L106). StructuredCoder doesn't enforce overriding getEncodedElementByteSize so I'm not sure if this is best practice. Ideally I would like to loop and invoke this method for every coder on the list. If underlying coder has default implementation, it will just serialize as is.

Reply:
I would remove the typecheck on StructuredCoder, just loop and add for every coder. The only downside is that there may be overhead constructing multiple CountingOutputStreams rather than creating one, but as you say there's no contract between StructuredCoder and implementing getEncodedElementByteSize. 

Go ahead and remove the check in LengthPrefixCoder too, there's no benefit for ever skipping this attempt.

Reply:
@robertwb thanks, changed.